app/support/holiday_validations.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module HolidayValidations
+
+  extend ActiveSupport::Concern
+
+  included do
+    validate :holiday_accessible, if: :starts_on_holiday?
+  end
+
+  def holiday_accessible
```

Comment:
The way that this concern is written, adds a lot of methods to the including-object.

It does not appear that you actually want/need to expose `holiday_accessible`, `user_restricted_on_holidays?`, etc to any users of the including object.

I propose extracting this logic into a real PORO.

doing so would have these benefits:
- Explicitly make clear which attributes of the including object are being used
- Allow for easier testing, by making the attributes explicit.

Example:

```ruby
module HolidayValidations
  extend ActiveSupport::Concern

  included do
    validate :holiday_accessible
  end

  def holiday_accessible
    # this makes explicit which attributes you are depending on
    holiday_validator = HolidayValidator.new(
                                        product: product,
                                        user: user
                                        event_start_date: event_start_date

    return unless holiday_validator.restricted?

    errors.add(:base, :holiday_access_restricted)
end

class HolidayValidator
  ....
  def restricted?
    return false unless starts_on_holiday?

    product.restrict_holiday_access? && user_restricted_on_holidays?
  end
  ...
end
```